modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.api;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.ignite.internal.app.IgniteImpl;
+import org.apache.ignite.internal.sql.engine.AbstractBasicIntegrationTest;
+import org.apache.ignite.internal.sql.engine.SqlQueryProcessor;
+import org.apache.ignite.internal.sql.engine.session.SessionId;
+import org.apache.ignite.internal.sql.engine.session.SessionManager;
+import org.apache.ignite.internal.testframework.IgniteTestUtils;
+import org.apache.ignite.lang.IgniteException;
+import org.apache.ignite.sql.IgniteSql;
+import org.apache.ignite.sql.Session;
+import org.junit.jupiter.api.Test;
+
+/** Test common SQL API. */
+public class ItCommonApiTest extends AbstractBasicIntegrationTest {
+    protected SqlQueryProcessor queryProcessor() {
+        return (SqlQueryProcessor)((IgniteImpl) CLUSTER_NODES.get(0)).queryEngine();
+    }
+
+    /**
+     * Gets the SQL API.
+     *
+     * @return SQL API.
+     */
+    protected IgniteSql igniteSql() {
+        return CLUSTER_NODES.get(0).sql();
+    }
+
+    @Override protected int nodes() {
+        return 1;
+    }
+
+    /** Check correctness of session expiration. */
+    @Test
+    public void testSessionExpiration() throws Exception {
+        long timeout = TimeUnit.SECONDS.toMillis(10); // time from SessionManager.checkPeriod
+
+        IgniteSql sql = igniteSql();
+
+        var queryProc = queryProcessor();
+
+        SessionManager sessionManager = IgniteTestUtils.getFieldValue(queryProc, "sessionManager");
```

Comment:
We need to stop messing around with component internals in e2e tests. I believe this test could be rewritten in order to use QueryCursor

Reply:
agree, done